osquery/tables/system/windows/shortcut_files.cpp

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/core/core.h>
+#include <osquery/core/tables.h>
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/tables/system/windows/registry.h>
+#include <osquery/utils/conversions/join.h>
+
+#include <osquery/logger/logger.h>
+#include <osquery/utils/windows/shelllnk.h>
+
+#include <boost/filesystem.hpp>
+
+#include <sstream>
+#include <string>
+
+namespace osquery {
+namespace tables {
+
+QueryData genShortcutFiles(QueryContext& context) {
+  QueryData results;
+  auto paths = context.constraints["path"].getAll(EQUALS);
+  // Expand contraints
+  context.expandConstraints(
+      "path",
+      LIKE,
+      paths,
+      ([&](const std::string& pattern, std::set<std::string>& out) {
+        std::vector<std::string> patterns;
+        auto status =
+            resolveFilePattern(pattern, patterns, GLOB_ALL | GLOB_NO_CANON);
+        if (status.ok()) {
+          for (const auto& resolved : patterns) {
+            out.insert(resolved);
+          }
+        }
+        return status;
+      }));
+
+  boost::system::error_code ec;
+  for (const auto& lnk : paths) {
+    boost::filesystem::path path = lnk;
+    if (!boost::filesystem::is_regular_file(path, ec)) {
+      continue;
+    }
+    std::string lnk_content;
+    if (!readFile(path, lnk_content, 20).ok()) {
+      LOG(WARNING) << "Failed to read shortcut file: " << lnk;
```

Comment:
so i hope this does not fail often lol
but since a path is required for input, im not sure what files will be read
so there is a chance readFile could fail (ex: a file is locked)
ive never seen a shortcut file be locked, but other files in the path may be

i switched to VLOG(1)